kwave/data.py

```diff
@@ -1,10 +1,55 @@
 from dataclasses import dataclass
+from typing import Any
 
 import numpy as np
 
 
+class Vector(np.ndarray):
+    def __new__(cls, elements: list):
+        elements = list(elements)[:3]  # Truncate the input list to have at most 3 elements
+        obj = np.array(elements).view(cls)
+        return obj
+
+    @property
+    def x(self):
+        return self[0]
+
+    @x.setter
+    def x(self, value):
+        self[0] = value
+
+    @property
+    def y(self):
+        if self.shape[0] > 1:
+            return self[1]
+        return np.nan
+
+    @y.setter
+    def y(self, value):
+        if self.shape[0] > 1:
+            self[1] = value
+
+    @property
+    def z(self):
+        if self.shape[0] > 2:
+            return self[2]
+        return np.nan
+
+    @z.setter
+    def z(self, value):
+        if self.shape[0] > 2:
+            self[2] = value
+
+    def assign_dim(self, dim: int, val: Any):
+        self[dim - 1] = val
+
+    def append(self, value):
+        new_coordinates = list(self) + [value]
+        return Vector(new_coordinates)
+
+
 @dataclass
-class Array(object):
+class FlexibleVector(object):
     """
         This class is very similar to Numpy.ndarray but there are differences:
             - It can have 3 elements at max
```

Comment:
Add comment that this class will be deprecated after kGrid is refactored.

Reply:
:+1: Done!